File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> values;

    for (int k = 0; k < nums.size(); ++k) {
        int min = target - nums.at(k);
        auto mFind = find(values.begin(), values.end(), nums.at(k));
        if (mFind != values.end()) {
            int index = mFind - values.begin();
            return {index, k};
        }

        values.push_back(min);
    }

    return {};
    }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
	var mmap = make(map[int]int)

	for i := 0; i < len(nums); i++ {
		// Cek apakah elemen dari "nums" di index ke-i ada di dalam "mmap"
		if _, contains := mmap[nums[i]]; contains {
			return []int{mmap[nums[i]], i}
		} else {
			mmap[target-nums[i]] = i
		}
	}
	
	return []int {}
}